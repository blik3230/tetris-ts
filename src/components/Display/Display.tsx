import React from 'react';
import './Display.css';

interface Prop {
  screenData?: Array<boolean[]>;
}

const Display = (props: Prop) => {
  const { screenData = [] } = props;

  return (
    <div className="Display">
      {
        screenData.map((lineItems, lineIndex) => {

          return (
            <div className="Display__line" key={lineIndex}>
              {
                lineItems.map((itemValue, itemIndex) => {

                  return (
                    <div
                      key={itemIndex}
                      className={`Display__item${itemValue ? " Display__item_active" : ""}`}
                    />
                  )
                })
              }
            </div>
          );
        })
      }
    </div>
  )
};

export default Display;