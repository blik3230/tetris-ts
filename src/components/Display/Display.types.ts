import { bitmapItemEnum } from '../../hooks/useTetris/types';

export type Bitmap = bitmapItemEnum[][];