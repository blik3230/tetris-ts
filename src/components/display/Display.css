.Display {
  display: inline-block;
  border: 2px solid #c9c9c9;
  border-radius: 4px;
}

.Display__line {
  display: flex;
}

.Display__item {
  box-sizing: border-box;
  width: 40px;
  height: 40px;
  margin: 1px;
}

.Display__item_active {
  border: 2px solid #c9c9c9;
  border-radius: 4px;
  padding: 2px;
}

.Display__item_active:after {
  content: '';
  display: block;
  width: 100%;
  height: 100%;
  background-color: #c9c9c9;
  border-radius: 4px;
}