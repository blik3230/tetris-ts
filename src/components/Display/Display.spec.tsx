import React from 'react';
import { render } from '@testing-library/react';
import Display from './Display';

describe('Display', () => {
  it('renders empty wrapper if received empty array by default', () => {
    const { asFragment } = render(<Display/>);

    expect(asFragment()).toMatchSnapshot();
  })

  it('renders grid of items from passed screenData array of boolean arrays', (done) => {
    const screenData = [
      [false, false, false, false, false, false],
      [false, false, false, false, false, false],
      [false, false, false, false, false, false],
      [false, false, false, false, false, false],
      [false, false, false, false, false, false],
      [false, false, false, false, false, false],
    ];

    const { container } = render(<Display screenData={ screenData }/>);

    const displayEl = container.querySelector('.Display')!;

    const lineEls = displayEl.querySelectorAll('.Display__line');

    expect(lineEls.length).toBe(6);

    for (const lineEl of lineEls) {
      const items = lineEl.querySelectorAll('.Display__item');

      expect(items.length).toBe(6);
    }

    done();
  });

  it('renders active items where items are truly', () => {
    const screenData = [
      [true, false, false, false, false],
      [true, true, false, false, false],
      [true, true, true, false, false],
      [true, true, true, true, false],
      [true, true, true, true, true],
    ];

    const { container } = render(<Display screenData={ screenData }/>);

    const displayEl = container.querySelector('.Display')!;
    const lineEls = displayEl.querySelectorAll('.Display__line')!;
    const expectedArr = [];

    for(const lineEl of lineEls) {
      const lineArr = [];
      const lineItemEls = lineEl.querySelectorAll('.Display__item');
      for(const itemEl of lineItemEls) {
        lineArr.push(itemEl.classList.contains('Display__item_active'));
      }
      expectedArr.push(lineArr);
    }

    expect(expectedArr).toEqual(screenData);
  });
});