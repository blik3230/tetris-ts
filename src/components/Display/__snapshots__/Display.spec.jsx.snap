// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders active items where items are truly 1`] = `
<DocumentFragment>
  <div
    class="Display"
  >
    <div
      class="Display__line"
    >
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
    </div>
    <div
      class="Display__line"
    >
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
    </div>
    <div
      class="Display__line"
    >
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
    </div>
    <div
      class="Display__line"
    >
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
    </div>
    <div
      class="Display__line"
    >
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
      <div
        class="Display__item"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Display renders empty wrapper if received empty array by default 1`] = `
<DocumentFragment>
  <div
    class="Display"
  />
</DocumentFragment>
`;
